### ISSUE TITLE
Animation group negative speed ratio is not working.

### ISSUE THREAD

**Issue by sebavan** [BabylonJS/Babylon.js#11225]
Following this thread: https://forum.babylonjs.com/t/repeated-animationgroup-onanimationloopobservable-triggered-when-speedratio-0/24361

We are facing two issues:

1. The onAnimationLoopObservable keeps triggering addressed by #11224 

2. The animation does not reverse correctly :
  a. You can see in this PG the scale going out of the animation bounds: https://playground.babylonjs.com/#CBGEQX#656
![image](https://user-images.githubusercontent.com/16655165/136294705-5862fafb-55bd-48b1-928b-6c952ad05d0e.png)

  b. Also on rotation only, the rotation does not seem to reverse from its current location: https://playground.babylonjs.com/#CBGEQX#659

2. a. seems to be coming from animationGroup Start function having the following code
```
this._speedRatio = speedRatio;

if (from !== undefined && to !== undefined) {
    if (from < to && this._speedRatio < 0) {
        let temp = to;
        to = from;
        from = temp;
    } else if (from > to && this._speedRatio > 0) {
        this._speedRatio = -speedRatio;
    }
}
```

This block should probably be placed before starting the animatable to take in account the new values for from and to :-) and should probably be running no matter if the parameters are passed in or if we use the default from and to.

2. b. is probably coming from a new animation running not accounting for the current value.

**A nice to have would be able to dynamically update the speedRatio without restarting the animation :-)**
